# Single-Cell RNA-Sequencing Technologies![image](https://user-images.githubusercontent.com/31963567/117818342-825c5780-b236-11eb-8137-004510bef66d.png)

# Plate-based SMART-seq: flow cytometry sort cells into 96 well plate, lyse cell and reverse transcribe, poly A based, additional CCC added to the end of reverse transcribed strand, template switch oligo with GGG anneal and elongate, PCR amplification. Library prep: transposes fragment and add adaptors, PCR and sequencing.

# DropSeq: microfluidics, beads, cell, oil, aqueous phase form emulsion, each beads has a unique beads barcode and barcoded oligos on the surface (UMI), capture transcripts, pull down beads, reverse transcription with template switching process.

# 10X Genomics: similar to DropSeq, high single beads and single cell rate due to the manipulation of the beads and microfluids machine, both capture and reverse transcription happens in the emulsion bubble. Break the emulsion and cDNA amplification for lib prep. 

# Microwells: SeqWell, beads in each well, drop cells in each well, lyseâ€¦

#cCombinatorial Indexing: SPLiT-seq, multiple cells in one well, multiple rounds of PCR and indexing.
inDrop is used for high-throughput single-cell labeling. This approach is similar to Drop-seq, but it uses hydrogel microspheres to introduce the oligonucleotides.
![image](https://user-images.githubusercontent.com/31963567/117818376-89836580-b236-11eb-9a0c-338e14c2a811.png)
